packages/components/src/focal-point-picker/index.js

```diff
@@ -0,0 +1,251 @@
+	}
+	pickerDimensions() {
+		if ( this.containerRef.current ) {
+			return {
+				width: this.containerRef.current.clientWidth,
+				height: this.containerRef.current.clientHeight,
+				top: this.containerRef.current.getBoundingClientRect().top + document.body.scrollTop,
+				left: this.containerRef.current.getBoundingClientRect().left,
+			};
+		}
+		return {
+			width: 0,
+			height: 0,
+			left: 0,
+			top: 0,
+		};
+	}
+	handleFocusOutside() {
+		this.setState( {
+			isDragging: false,
+		} );
+	}
+	render() {
+		const { instanceId, url, value, label, help, className } = this.props;
+		const { bounds, isDragging, percentages } = this.state;
+		const pickerDimensions = this.pickerDimensions();
+		const iconCoordinates = {
+			left: ( value.x * ( pickerDimensions.width - ( bounds.left * 2 ) ) ) + bounds.left,
+			top: ( value.y * ( pickerDimensions.height - ( bounds.top * 2 ) ) ) + bounds.top,
+		};
+		const iconContainerStyle = {
+			left: `${ iconCoordinates.left }px`,
+			top: `${ iconCoordinates.top }px`,
+		};
+		const iconContainerClasses = classnames(
+			'components-focal-point-picker__icon_container',
+			isDragging ? 'is-dragging' : null
+		);
+		const id = `inspector-focal-point-picker-control-${ instanceId }`;
+		const horizontalPositionId = `inspector-focal-point-picker-control-horizontal-position-${ instanceId }`;
+		const verticalPositionId = `inspector-focal-point-picker-control-horizontal-position-${ instanceId }`;
+		return (
+			<BaseControl label={ label } id={ id } help={ help } className={ className }>
+				<div className="components-focal-point-picker-wrapper">
+					<div
+						className="components-focal-point-picker"
+						onMouseDown={ () => this.setState( { isDragging: true } ) }
+						onDragStart={ () => this.setState( { isDragging: true } ) }
+						onMouseUp={ () => this.setState( { isDragging: false } ) }
+						onDrop={ () => this.setState( { isDragging: false } ) }
+						onMouseMove={ this.onMouseMove }
+						ref={ this.containerRef }
+						role="button"
+						tabIndex="-1"
+					>
+						<img
+							alt="Dimensions helper"
+							onLoad={ this.onLoad }
+							ref={ this.imageRef }
+							src={ url }
+							draggable="false"
+						/>
+						<div className={ iconContainerClasses } style={ iconContainerStyle }>
+							<SVG className="components-focal-point-picker__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
+								<Path className="components-focal-point-picker__icon-outline" d="M15 1C7.3 1 1 7.3 1 15s6.3 14 14 14 14-6.3 14-14S22.7 1 15 1zm0 22c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8z" />
+								<Path className="components-focal-point-picker__icon-fill" d="M15 3C8.4 3 3 8.4 3 15s5.4 12 12 12 12-5.4 12-12S21.6 3 15 3zm0 22C9.5 25 5 20.5 5 15S9.5 5 15 5s10 4.5 10 10-4.5 10-10 10z" />
+							</SVG>
+						</div>
+					</div>
+				</div>
+				<div className="components-focal-point-picker_position-display-container">
+					<BaseControl label={ __( 'Horizontal Pos.' ) }>
```

Comment:
Per @afercia 's comment, I think this needed to have been assigned an `id` for a label to be produced associated with the rendered `input` (which _does_ have an `id` associated).

It seems at some point `BaseControl` was updated to be accommodating to receiving a `label` and not an `id`, at which point it renders a `span`. I don't know that this should be supported, as it leaves open the possibility for errors like this.

https://github.com/WordPress/gutenberg/blob/858719bf3e5c7cd37c14824f5f3d4b8db49cc8f2/packages/components/src/base-control/index.js#L11

Reply:
Thanks @aduth I think that change was made for the new font size picker:

<img width="447" alt="screenshot 2019-02-15 at 15 23 32" src="https://user-images.githubusercontent.com/1682452/52862544-cb93ce80-3135-11e9-8b83-6e32cbc08ef8.png">

where the visible text can't be a label element. At that point though, the input child _must_ have an `aria-label` attribute: and yes, this leaves open the possibility that developers forget or don't know they need to label the input.

Reply:
> where the visible text can't be a label element

(Related: https://github.com/WordPress/gutenberg/pull/9802#discussion_r257250785 )

Forgive my understanding, but it's not clear to me why in that example "Font Size" would not be a label element associated with the dropdown input?

Reply:
Not all elements can have an associated <label> element. Form elements and a few others can. What is the dropdown input in this case? The div?

Reply:
> Not all elements can have an associated element. Form elements and a few others can. What is the dropdown input in this case? The div?

There's a `button` which could receive the initial focus for the dropdown.

![image](https://user-images.githubusercontent.com/1779930/52866415-9921ab80-310c-11e9-9344-a1905b8f7239.png)

Reply:
Yes, technically a button is a labelable element, here's a list: https://www.w3.org/TR/html52/sec-forms.html#labelable-element

However, it's not just about click > focus. The primary goal of labels is giving names to controls. In this case, it wouldn't make sense. A `<label>` element would override the button text. It wouldn't override the button aria-label though, which is, for example: `aria-label: "Font size: Normal"`. It would be a very uncommon usage. Seems to me your proposal in https://github.com/WordPress/gutenberg/pull/9802#discussion_r257269125 is preferable.

Reply:
FYI these specific `BaseControl` were corrected by @jorgefilipecosta in #14152